solr/core/src/java/org/apache/solr/query/FilterQuery.java

```diff
@@ -96,6 +95,6 @@ public Weight createWeight(IndexSearcher searcher, ScoreMode scoreMode, float bo
     DocSet docs = solrSearcher.getDocSet(q);
     // reqInfo.addCloseHook(docs);  // needed for off-heap refcounting
 
-    return new BoostQuery(new SolrConstantScoreQuery(docs.makeQuery()), 0).createWeight(searcher, scoreMode, 1f);
+    return new BoostQuery(new ConstantScoreQuery(docs.makeQuery()), 0).createWeight(searcher, scoreMode, 1f);
```

Comment:
Why boost a constant scoring query where we pick the score? :-)

Reply:
Thanks for pointing that out. I've changed it to boost the query in FilterQuery